src/fields/field_fe.cc

```diff
@@ -249,13 +315,15 @@ void FieldFE<spacedim, Value>::cache_update(FieldValueCache<typename Value::elem
             cell = dh_->cell_accessor_from_element( elm_idx );
             loc_dofs = cell.get_loc_dof_indices();
             last_element_idx = elm_idx;
+            range_bgn = this->fe_item_[elm.dim()].range_begin_;
+            range_end = this->fe_item_[elm.dim()].range_end_;
         }
 
         unsigned int i_ep=cache_map.eval_point_data(i_data).i_eval_point_;
         //DHCellAccessor cache_cell = cache_map(cell);
         mat_value.fill(0.0);
-        for (unsigned int i_dof=0; i_dof<loc_dofs.n_elem; i_dof++) {
-            mat_value += data_vec_[loc_dofs[i_dof]] * this->handle_fe_shape(cell.dim(), i_dof, i_ep, comp_index_);
+        for (unsigned int i_dof=range_bgn; i_dof<range_end; i_dof++) {
```

Comment:
Fix and test error with passing i_dof to handle_fe_shape. Need separate index starting from 0.